File: kt/src/main/kotlin/com/github/scglwsj/parking/lot/Ticket.kt
package com.github.scglwsj.parking.lot

import java.util.*

data class Ticket(private val carPlate: PlateNumber, val parkingLotID: ParkingLotID) {
    val id = TicketID(UUID.randomUUID().toString())
    var isValid = true
        private set

    fun invalidate() {
        isValid = false
    }
}

data class TicketID(val id: String)

File: ts/src/parking/domain/parkingLot/parking/ticket.ts
/** @format */

import {v4 as uuid} from 'uuid'
import {PlateNumber} from './car'
import {ParkingLotId} from './parkingLot'
import {ValueObject} from '../../valueObject'

export class Ticket extends ValueObject<TicketID> {
    private valid: Boolean

    public constructor(public readonly carPlante: PlateNumber, public readonly ParkingLot: ParkingLotId) {
        super(new TicketID(uuid()))
        this.valid = true
    }

    public get isValid() {
        return this.valid
    }

    public get id() {
        return this.value
    }

    public invalidate() {
        this.valid = false
    }
}

export class TicketID extends ValueObject<String> {}
